### PR TITLE
BuildMojo - Delete .jar.original for both file and directory case

### LINKED ISSUE #12786: devtools/cli generates quarkus-cli-999-SNAPSHOT.jar.original as directory and BuildMojo code can't handle it
devtools/cli generates quarkus-cli-999-SNAPSHOT.jar.original as directory and BuildMojo code can't handle it

To reproduce just call twice `mvn package -pl devtools/cli/ -DskipTests`. Important piece is that `clean` is not part of the command.

mvn command ends with:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project 
 quarkus-cli: Failed to build quarkus application: java.nio.file.DirectoryNotEmptyException: /Users/rsvoboda/tmp/
 quarkus/devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original -> [Help 1]
```

Details from `mvn -X ...` execution
```
Caused by: java.nio.file.DirectoryNotEmptyException: /Users/rsvoboda/tmp/quarkus/devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original
    at sun.nio.fs.UnixFileSystemProvider.implDelete (UnixFileSystemProvider.java:247)
    at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists (AbstractFileSystemProvider.java:110)
    at java.nio.file.Files.deleteIfExists (Files.java:1180)
    at io.quarkus.maven.BuildMojo.doExecute (BuildMojo.java:100)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute (QuarkusBootstrapMojo.java:90)
...
```

Code in BuildMojo expects file, not directory like in devtools/cli case
```
$ find devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Clean.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/core
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/core/QuarkusVersion.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/core/BaseSubCommand.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/core/BuildsystemCommand.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/core/ExecuteUtil.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Remove.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Create$TargetLanguage.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/List$ExtensionFormat.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/QuarkusCli.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Add.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Dev.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/List.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Build.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/QuarkusCli$ExecutionStrategy.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Create.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/io/quarkus/cli/Create$TargetBuildTool.class
devtools/cli/target/quarkus-cli-999-SNAPSHOT.jar.original/application.properties
```

Probably related to https://github.com/quarkusio/quarkus/pull/12725 changes, CI job started to fail 4 days ago

### PR DESCRIPTION
**Pull request by rsvoboda**
BuildMojo - Delete .jar.original for both file and directory case

Fixes https://github.com/quarkusio/quarkus/issues/12786

@aloubyansky fyi

### CODE REVIEW

File: devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java [hunk 1]
```diff
@@ -97,7 +98,12 @@ protected void doExecute() throws MojoExecutionException {
                         final Path standardJar = curatedApplication.getAppModel().getAppArtifact().getPaths().getSinglePath();
                         if (Files.exists(standardJar)) {
                             try {
-                                Files.deleteIfExists(result.getJar().getOriginalArtifact());
+                                if (Files.exists(result.getJar().getOriginalArtifact())) {
+                                    Files.walk(result.getJar().getOriginalArtifact())
+                                            .sorted(Comparator.reverseOrder())
+                                            .map(Path::toFile)
+                                            .forEach(File::delete);
+                                }
```
Comment (aloubyansky) [inline devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:106]: Perhaps we should call https://github.com/quarkusio/quarkus/blob/master/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/util/IoUtils.java#L58 instead as this is what we've been using everywhere until now.

Comment (rsvoboda) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:106]: Ok, thanks forpoinnter, I will update the PR

**Comment by rsvoboda** [comment]
@aloubyansky updated

**Review by aloubyansky — APPROVED** [review]
Thanks @rsvoboda !

**Comment by rsvoboda** [comment]
Timeout on Windows run doesn't look related to this PR. The rest is green

**Pull request merged**